cmd/ipfs/daemon.go

```diff
@@ -109,6 +113,14 @@ func daemonFunc(req cmds.Request, res cmds.Response) {
 		}
 	}()
 
+	// check transport encryption flag.
+	unencrypted, _, _ := req.Option(unencryptTransportKwd).Bool()
+	if unencrypted {
+		log.Warningf(`Running with --%s: All connections are UNENCRYPTED.
+		You will not be able to connect to regular encrypted networks.`, unencryptTransportKwd)
+		conn.EncryptConnections = false
```

Comment:
Exposed package globals like this frighten me. Will we remove this once node interop works?

If it stays around, I'd prefer `conn.DisableEncryption()` and print a warning on each dial as well.